fhevm/tee_pad_gas_test.go

```diff
@@ -0,0 +1,32 @@
+package fhevm
+
+import (
+	"testing"
+
+	"github.com/holiman/uint256"
+	"pgregory.net/rapid"
+)
+
+func TestTeePadRequiredGas(t *testing.T) {
+	rapid.Check(t, func(t *rapid.T) {
+		testcases := []struct {
+			suppliedGas uint64
+			padTo       uint256.Int
+			expected    uint64
+		}{
+			{40, *uint256.NewInt(20), 0},
+			{90, *uint256.NewInt(110), 90},
```

Comment:
here, the tc.expected shouldn't be 90. It should be 110

Reply:
Since consumedGas = 100 (```gasLimit```) - 90 (```suppliedGas```) = 10, we'd expect the function to return 110 (```padTo```) - 10 (```consumedGas```) which is 100?

Reply:
True!